src/compiler/binder.ts

```diff
@@ -460,6 +462,11 @@ namespace ts {
                     // A symbol already exists, so don't add this as a declaration.
                     return symbol;
                 }
+                else if(!isReplaceableByMethod && symbol.isReplaceableByMethod) {

```

Comment:
this doesn't seem right. I thought the existing replaceable-by-method code in declareSymbol would be good enough for computed names too.

Reply:
This fixes the types and symbols on tests allowJscheckJsTypeParameterNoCrash`. If you feel like this are wrong I can roll it back as without this the crash is fixed.

Reply:
I vote to roll it back, since that's an incorrect JS program, so there's no real requirement to get binding correct. And we already issue a correct error, so people will be led to the fix.

Reply:
Sounds good, done.